src/klein/_app.py

```diff
@@ -91,12 +77,41 @@ def __call__(
         """
 
 
-KleinRouteHandler = Union[KleinRouteFunction, KleinRouteMethod]
+KleinRouteHandler = Callable[..., KleinRenderable]
+"""
+A handler for a Klein route.  Sadly it is not possible to tell the type checker
+that the first argument is IRequest or the first/second arguments are (Self,
+IRequest); this is not a useful actual bound on the type in terms of what you
+can do with it, so Mypy can't conceptualize it; the rest of the signature is a
+Werkzeug/@require-derived soup.  So "you didn't pass a request" is just an
+informative message for the user, not a type assertion; checking the parameter
+list will have to be handled at runtime.
+"""
+
+SomeKleinHandler = TypeVar("SomeKleinHandler", bound=KleinRouteHandler)
```

Comment:
I think `KleinHandlerT` or `KleinRouteHandlerT` would be a more standard name for this?

Reply:
I'll do a rename pass on all of these; as the current naming may suggest, these were selected after a long and frustrating crash against every branch of the confused-about-typing tree as I hurtled towards the ground.

Reply:
Oh hm I guess that's the only really crummy name in here. Great!